code/app/src/main/java/com/adobe/marketing/edge/identity/app/ui/CustomIdentityFragment.kt

```diff
@@ -93,6 +105,44 @@ class CustomIdentityFragment : Fragment() {
             Identity.removeIdentity(item, namespace)
         }
 
+        // Button for Set Ad ID behavior
+        // Sets the advertising identifier set in the corresponding textfield using the MobileCore API
+        root.findViewById<Button>(R.id.btn_set_ad_id).setOnClickListener {
+            val adId: String? = sharedViewModel.adId.value
+            MobileCore.setAdvertisingIdentifier(adId)
+        }
+
+        // For details on implementation tips and differences between Google Play Services Ads vs AndroidX Ads,
+        // please see the project Documentation -> AEPEdgeIdentity.md : Test App -> GAID Implementation
+        root.findViewById<Button>(R.id.btn_get_gaid).setOnClickListener {
+            val context = context
+            Log.d(LOG_TAG, "context: $context, appContext: ${context?.applicationContext}")
+            if (context != null) {
+                // Implementation for google play services ads
+                // Logic implemented in ViewModel to leverage managed Kotlin coroutine scope from lifecycle-viewmodel-ktx
+                // lifecycle-viewmodel-ktx requires compiled sdk version 31+
+                GlobalScope.launch {
+                    sharedViewModel.getGAID(context.applicationContext)
+                }
+            }
+        }
+
+        root.findViewById<Button>(R.id.btn_set_ad_id_null).setOnClickListener {
+            Log.d(LOG_TAG, "Setting advertising identifier to: null")
+            MobileCore.setAdvertisingIdentifier(null)
+        }
+
+        root.findViewById<Button>(R.id.btn_set_ad_id_all_zeros).setOnClickListener {
+            Log.d(LOG_TAG, "Setting advertising identifier to: 00000000-0000-0000-0000-000000000000")
+            MobileCore.setAdvertisingIdentifier("00000000-0000-0000-0000-000000000000")
```

Comment:
nit: can use a constant for ALL_ZEROES_ADID and reuse it in logs too

Reply:
Usage updated!